### ISSUE TITLE
Bug/Question - Possible Circular Dependency.

### ISSUE THREAD

**Issue by jcperez-ch** [quasarframework/quasar#2444]
### Software version

Quasar: 0.17.9
OS: Mac OS
Node: 10.0
NPM: 6.3.0
Browsers: Chrome/Safari
Any other software related to your bug:

### What were you expecting?
Normal behaviour when not importing the QSpinner, since it's a dependency of QInput, either that or update the docs to add manually the dependency to QSpinner

### What steps did you take, to get the error?
After installing a brand new project using the Vue Cli 3, or updating to version 0.17.9 and adding the QSpinner as [shown in the docs](https://quasar-framework.org/components/spinner.html), Vue complains as:

```
[Vue warn]: Error in render: "TypeError: Cannot read property 'components' of undefined"

found in

---> <QInput>
```

https://github.com/quasarframework/quasar/blob/3cc189d268ae88d0b11a6dcc8c922f4ab67f41d6/src/components/spinner/QSpinner.js#L1

@rstoenescu isn't this a circular dependency?

**Comment by lesharris** [comment]
I just came here to post the same issue.  In my case my existing application no longer loads when the QSpinner component is being registered during Vue instantiation.  It eventually dies during checkCompnents in the vue runtime:

```javascript
/**
 * Validate component names
 */
function checkComponents (options) {
  for (var key in options.components) { // Dies here.  options is null for QSpinner
    validateComponentName(key);
  }
}
```

**Comment by rstoenescu** [comment]
Hi,

Please redirect this to the appropriate repo: [vue-cli-plugin-quasar](https://github.com/quasarframework/vue-cli-plugin-quasar)

Also, what version of vue-cli-plugin-quasar are you using? Is it vue-cli-plugin-quasar v0.17.0 (which is latest)?

Thanks.

**Comment by jcperez-ch** [comment]
@rstoenescu issue created under https://github.com/quasarframework/vue-cli-plugin-quasar/issues/8

@lesharris the workaround is to modify your `vue.config.js` and add this:
```
module.exports = {
    // ... more options
    chainWebpack: config => {
        config.resolve.extensions.prepend('.mat.js')
    },
```

**Comment by rstoenescu** [comment]
There was code dealing with it, but needed tweaking. Released vue-cli-plugin-quasar v0.17.1 having this fixed. Thanks for reporting!
